src/main/java/org/testng/internal/Utils.java

```diff
@@ -155,11 +153,13 @@ public static void writeFile(@Nullable String outputDir, String fileName, String
    * Writes the content of the sb string to the file named filename in outDir. If
    * outDir does not exist, it is created.
    *
-   * @param outDir the output directory (may not exist). If <tt>null</tt> then current directory is used.
-   * @param fileName the filename
+   * @param outputFolder the output directory (may not exist). If <tt>null</tt> then current directory is used.
+   * @param filename the filename
    * @param sb the file content
    */
-  private static void writeFile(@Nullable File outDir, String fileName, String sb, @Nullable String encoding) {
+  private static void writeFile(@Nullable File outputFolder, String filename, String sb, @Nullable String encoding) {
+    File outDir = outputFolder;
```

Comment:
The general rule is that method parameters are not supposed to be altered within a method but they should instead be copied to a different variable and then operated upon.